### PR TITLE
add WECHAT_MINI_PROGRAM macro

### PR DESCRIPTION
**Pull request by novain1102**
Re: #

### Changelog

*

-------

### Continuous Integration

This pull request:

* [ ] needs automatic test cases check.
  > Manual trigger with `@cocos-robot run test cases` afterward.
* [ ] does not change any runtime related code or build configuration
  > If any reviewer thinks the CI checks are needed, please uncheck this option, then close and reopen the issue.

-------

### Compatibility Check

This pull request:

* [ ] changes public API, and have ensured backward compatibility with [deprecated features](https://github.com/cocos/cocos-engine/blob/v3.5.0/docs/contribution/deprecated-features.md).
* [ ] affects platform compatibility, e.g. system version, browser version, platform sdk version, platform toolchain, language version, hardware compatibility etc.
* [ ] affects file structure of the build package or build configuration which requires user project upgrade.
* [ ] **introduces breaking changes**, please list all changes, affected features and the scope of violation.

### CODE REVIEW

File: cocos/asset/assets/image-asset.ts [hunk 1]
```diff
@@ -24,7 +24,7 @@
 
 // @ts-check
 import { ccclass, override } from 'cc.decorator';
-import { EDITOR, ALIPAY, XIAOMI, JSB, TEST, BAIDU, TAOBAO, TAOBAO_MINIGAME, WECHAT } from 'internal:constants';
+import { EDITOR, ALIPAY, XIAOMI, JSB, TEST, BAIDU, TAOBAO, TAOBAO_MINIGAME, WECHAT_MINI_PROGRAM } from 'internal:constants';
```
Comment (minggo) [inline cocos/asset/assets/image-asset.ts:27]: Don‘t need `WECHAT`?

File: cocos/asset/assets/image-asset.ts [hunk 2]
```diff
@@ -163,7 +163,7 @@ function fetchImageSource (imageSource: ImageSource) {
 
 // 返回该图像源是否是平台提供的图像对象。
 function isNativeImage (imageSource: ImageSource): imageSource is (HTMLImageElement | HTMLCanvasElement | ImageBitmap) {
-    if (ALIPAY || TAOBAO || TAOBAO_MINIGAME || XIAOMI || BAIDU || WECHAT) { // TODO(xwx):WECHAT only for wechat miniprogram, need to refactor
+    if (ALIPAY || TAOBAO || TAOBAO_MINIGAME || XIAOMI || BAIDU || WECHAT_MINI_PROGRAM) {
```
Comment (minggo) [inline cocos/asset/assets/image-asset.ts:166]: The same as above.

File: cocos/asset/assets/image-asset.ts (hunk 1 above)
Comment (novain1102) [reply cocos/asset/assets/image-asset.ts:27]: Yes. Use `WECHAT` before just to hack.

File: cocos/asset/assets/image-asset.ts (hunk 2 above)
Comment (novain1102) [reply cocos/asset/assets/image-asset.ts:166]: This logic only for wechat mini program(WECHAT_MINI_PROGRAM). not for wechat minigame(WECHAT)

**Review by minggo — APPROVED** [review]

**Review by PPpro — APPROVED** [review]

**Pull request merged**